src/pages/ContributorFields.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TextField, Typography } from '@material-ui/core';
+import { Contributor, Resource } from '../types/resource.types';
+import { ErrorMessage, Field, FieldArray, FieldProps, FormikProps, FormikValues } from 'formik';
+import Button from '@material-ui/core/Button';
+import { createContributor, putContributorFeature } from '../api/resourceApi';
+import Paper from '@material-ui/core/Paper';
+import styled from 'styled-components';
+import DeleteIcon from '@material-ui/icons/Delete';
+import AddIcon from '@material-ui/icons/Add';
+
+const StyledPaper = styled(Paper)`
+  width: 100%;
+  padding: 3rem;
+`;
+
+const StyledField = styled(Field)`
+  display: inline-block;
+  margin-left: 2rem;
+  margin-right: 2rem;
+  padding: 1rem;
+`;
+const StyledDiv = styled.div`
+  display: inline-block;
+  margin-top: 1rem;
+  margin-bottom: 0.3rem;
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 22rem;
+  margin-left: 0.5rem;
+  margin-right: 0.5rem;
+`;
+
+const StyledButton = styled(Button)`
+  position: relative;
+  vertical-align: bottom;
+`;
+
+interface ContributorFieldsProps {
+  resource: Resource;
+  formikProps: FormikProps<FormikValues>;
+  setAllChangesSaved: any;
+}
+
+const ContributorFields: FC<ContributorFieldsProps> = ({ resource, formikProps, setAllChangesSaved }) => {
+  const { t } = useTranslation();
+
+  const addContributor = (arrayHelpers: any) => {
+    createContributor(resource.identifier).then((contributorResponse) => {
+      arrayHelpers.push({
+        identifier: contributorResponse.data.identifier,
+        features: {
+          dlr_contributor_name: '',
+          dlr_contributor_type: '',
+          dlr_contributor_identifier: contributorResponse.data.identifier,
+        },
+      });
+    });
+  };
+
+  const saveField = async (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>,
+    resetForm: any,
+    currentValues: FormikValues
+  ) => {
+    setAllChangesSaved(false);
+    const name = '' + event.target.name.split('.').pop();
+    if (event.target.value.length > 0) {
+      const indexArray = event.target.name.match(/\d+/);
+      let index = -1;
+      if (indexArray) {
+        index = parseInt(indexArray[0]);
+      }
```

Comment:
Ser ut som at `saveField` her bare brukes inni `FieldArray`. Hvis det stemmer kunne dere kanskje sendt inn `index` som argument, for den har dere jo når dere kaller `saveField()` i render, så slipper dere all parsing osv over her

Reply:
Jepp, så pcb hadde kommentert det samme. Er nå fikset :)

Reply:
wait hvorfor sender jeg ikke bare contributor identifieren avgårde istedet? Er jo den jeg bruker indexen til.